api/src/main/java/io/opencensus/trace/export/ActiveSpans.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace.export;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.trace.Span;
+import java.util.Collection;
+import java.util.Map;
+import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class allows users to access in-process information about all active spans.
+ *
+ * <p>The active spans tracking is available for all the spans with the option {@link
+ * Span.Options#RECORD_EVENTS}. This functionality allows users to debug stuck operations or long
+ * living operations.
+ */
+@ThreadSafe
+public abstract class ActiveSpans {
+
+  protected ActiveSpans() {}
+
+  /**
+   * Returns the number of active spans for every different span name.
+   *
+   * @return the number of active spans for every different span name.
+   */
+  public abstract Map<String, Integer> getNumberOfActiveSpans();
+
+  /**
+   * Returns a list of active spans that match the {@code Filter}.
+   *
+   * @param filter used to filter the returned spans.
+   * @return a list of active spans that match the {@code Filter}.
+   */
+  public abstract Collection<SpanData> getActiveSpans(Filter filter);
+
+  /**
+   * Filter for active spans. Used to filter results returned by the {@link #getActiveSpans(Filter)}
+   * request.
+   */
+  @AutoValue
+  @Immutable
+  public abstract static class Filter {
```

Comment:
For the same reasons as you pointed in the usage of classes for the return types I think it would be good to keep the name generic in case we want to add more conditions for filtering.